src/format.ts

```diff
@@ -0,0 +1,16 @@
+export function formatCost(value: number | undefined, precision = 4) {
+  if (value == null) {
+    return '?';
+  }
+
+  return `$${value.toFixed(precision)}`;
+}
+
+export function formatTokenCount(tokenCount: number, roundTo = 1) {
+  const roundedCount = Math.round(tokenCount / roundTo) * roundTo;
+
+  const suffixes = ['', 'K', 'M'];
+  const suffixIndex = Math.floor(Math.log10(Math.abs(roundedCount)) / 3);
```

Comment:
Thos calculations seem a bit complex. Can you add a comment or something?

Reply:
added overall comment. Tried adding line comments bu the